modules/osu_groups_basic_group/osu_groups_basic_group.module

```diff
@@ -116,3 +118,41 @@ function osu_groups_basic_group_block_view_system_branding_block_alter(array &$b
     'preRender',
   ];
 }
+
+/**
+ * Implements hook_form_BASE_FORM_ID_alter().
+ */
+function osu_groups_basic_group_form_node_form_alter(&$form, FormStateInterface $form_state) {
+  /** @var \Drupal\node\NodeInterface $node */
+  $node = $form_state->getFormObject()->getEntity();
+  // If node is new and created in group context
+  // OR node is a group page and does not have a normal menu entry
+  if (($form_state->get('group') && $node->isNew())
+      || (!menu_ui_get_menu_link_defaults($node)['id'] && isGroup())) {
+    $form['menu']['link']['menu_parent']['#default_value'] = 'Group Menu:';
+  }
+}
+
+/**
+  * Determines if existing page is a group page.
+  *
+  *
+  * @return bool
+  *   True if current page is a group page.
+  */
+function isGroup(): bool {
+  $is_group = FALSE;
+
+  $current_path = Drupal::service('path.current')->getPath();
+  $path = Drupal::service('path_alias.manager')->getPathByAlias($current_path);
+  if (preg_match('/node\/(\d+)/', $path, $matches)) {
+    $node = Node::load($matches[1]);
+    $osu_groups = new OsuGroupsHandler();
```

Comment:
This one is on me and I didn't create the OSU Groups Handler as a proper Drupal service for Dependency Injection. I have since made the correct modifications to the module and we should be avoiding using `new myObject` to instantiate new objects that are Drupal services and instead use the `\Drupal::service('id.of.service')` to load the object and have all possible dependencies wired in.

Change this to:
`$osu_groups = \Drupal::service('osu_groups.group_handler');`

If you need to find out what the ID of the service is you want to create you can use Drush with the Devel Module
`drush devel:services`
or pass a parameter to the end to try and filter down
`drush devel:services osu_groups` or any string to filter.